### PR TITLE
Deprecate ?demostorage in favour of demo://

### PR DESCRIPTION
**Pull request by navytux**
In 2021 in 2f02bc90 (demo: URI resolver) we added support for generic demo:// URI scheme which allows combining any two storages into a DemoStorage. This support is more generic because e.g. previously ?demostorage was provided only by file:// and zeo:// URI schemes, with the "changes" part there being hardcoded to in-memory MappingStorage.

Now there is no need to use ?demostorage option as we can use demo:// uniformly.

Deprecating ?demostorage in favour of dedicated DemoStorage resolver was actually suggested several times in the past:

https://github.com/Pylons/zodburi/pull/25#issuecomment-485506959  
https://github.com/Pylons/zodburi/pull/25#issuecomment-511480572  
https://github.com/Pylons/zodburi/pull/25#issuecomment-622931793  

/cc @azmeuk, @stevepiercy

### CODE REVIEW

**Comment by navytux** [comment]
P.S. @azmeuk, I appologize for my silence on https://github.com/Pylons/zodburi/pull/32.

**Review by stevepiercy — APPROVED** [review]
Docs like good to me. A maintainer should review the source code.

**Comment by navytux** [comment]
Thanks, @stevepiercy.

In the past I was [granted](https://github.com/Pylons/zodburi/pull/29#issuecomment-836034871) zodburi   maintaintership [together with @azmeuk](https://github.com/Pylons/zodburi/pull/29#issuecomment-836915440), so I think it should be enough if Éloi reviews this.

I would be glad to be proven wrong, but in my experience Jim, Tres and Chris are no longer active in around ZODB.

File: docs/index.rst
Comment (tseaver) [inline docs/index.rst:171]: Rather than removing the docs, please flag this option as deprecated (likewise below).

Comment (navytux) [reply docs/index.rst:171]: Thanks, @tseaver. I restored documentation and added corresponding deprecation notice to it in https://github.com/Pylons/zodburi/pull/33/commits/3298ac19.

Glad to see you around!

**Review by tseaver — APPROVED** [review]

**Comment by navytux** [comment]
Thanks, @tseaver.

**Comment by navytux** [comment]
With 2 approvals I think it is ok to merge this patch. I plan to do so in a day or two.

**Pull request merged**
